cli/find.php

```diff
@@ -0,0 +1,146 @@
+-h, --help                       Print out this help.
+
+Example:
+\$ sudo -u www-data /usr/bin/php admin/tool/advancedreplace/cli/find.php --search=thelostsoul --summary
+\$ sudo -u www-data /usr/bin/php admin/tool/advancedreplace/cli/find.php --regex-match=thelostsoul\\d+ --summary
+";
+
+list($options, $unrecognized) = cli_get_params(
+    [
+        'search'  => null,
+        'regex-match'  => null,
+        'tables' => '',
+        'summary' => false,
+        'help'    => false,
+    ],
+    [
+        'h' => 'help',
+    ]
+);
+
+if ($unrecognized) {
+    $unrecognized = implode("\n  ", $unrecognized);
+    cli_error(get_string('cliunknowoption', 'admin', $unrecognized));
+}
+
+// Ensure that we have required parameters.
+if ($options['help'] || (!is_string($options['search']) && empty($options['regex-match']))) {
+    echo $help;
+    exit(0);
+}
+
+// Ensure we only have one search method.
+if (!empty($options['regex-match']) && !empty($options['search'])) {
+    cli_error(get_string('errorsearchmethod', 'tool_advancedreplace'));
+}
+
+try {
+    if (!empty($options['search'])) {
+        $search = validate_param($options['search'], PARAM_RAW);
+    } else {
+        $search = validate_param($options['regex-match'], PARAM_RAW);
+    }
+    $tables = validate_param($options['tables'], PARAM_RAW);
+} catch (invalid_parameter_exception $e) {
+    cli_error(get_string('invalidcharacter', 'tool_advancedreplace'));
+}
+
+// Perform the search.
+$result = helper::search($search, !empty($options['regex-match']), $tables, $options['regex-match'] ? 1 : 0);
+
+// Notifying the user if no results were found.
+if (empty($result)) {
+    echo "No results found.\n";
+    exit(0);
+}
+
+// Show header.
+if (!$options['summary']) {
+    echo "Table, Column, ID, Match \n";
+} else {
+    echo "Table, Column\n";
+}
+
+// Output the result.
+foreach ($result as $table => $columns) {
+    foreach ($columns as $column => $rows) {
+        if ($options['summary']) {
+            echo "$table, $column\n";
+        } else {
+            foreach ($rows as $row) {
+                // Fields to show.
+                $id = $row->id;
+                $data = $row->$column;
+
+                if (!empty($options['regex-match'])) {
+                    // If the search string is a regular expression, show each matching instance.
+
+                    // Replace "/" with "\/", as it is used as delimiters.
+                    $search = str_replace('/', '\\/', $search);
+
+                    // Perform the regular expression search.
+                    preg_match_all( "/" . $search . "/", $data, $matches);
+
+                    if (!empty($matches[0])) {
+                        // Show the result foreach match.
+                        foreach ($matches[0] as $match) {
+                            echo "$table, $column, $id, \"$match\"\n";
```

Comment:
can you please use either the php or the moodle apis for writing csv so this is escaped properly for content when it contains commas and quotes